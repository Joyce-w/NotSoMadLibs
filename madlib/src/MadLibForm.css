.MadLibForm {
    margin: auto;
    display: flex;
    width: 25%;
    font-family:'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.MadLibForm-form input{
    padding: 15px;
    border: 1px solid #ccc;
    border-radius: 3px;
    margin-bottom: 10px;
    width: 70%;
    box-sizing: border-box;
    font-family: montserrat;
    color: #2C3E50;
    font-size: 13px;

}

.MadLibForm-form button {
    padding: 15px;
    border: 1px solid #ccc;
    border-radius: 3px;
    margin-bottom: 10px;
    width: 70%;
    box-sizing: border-box;
    font-family: montserrat;
    color: #2C3E50;
    font-size: 13px;
}
