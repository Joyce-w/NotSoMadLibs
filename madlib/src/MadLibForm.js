import React, { useEffect, useState } from "react"
import axios from "axios"
import { v4 as uuid } from "uuid"
import "./MadLibForm.css"

const MadLibForm = () => {

    const [madLib, setMadLib] = useState()
    const [data, setData] = useState([])

    //handle changed data
    const handleChange = (e) => {
        setData(data => ({
            ...data,
          [e.target.name]: e.target.value  
        }))
        
    }
    
    //get api data
    useEffect(() => {

        async function getMadLib () {
        const res = await axios.get('http://madlibz.herokuapp.com/api/random?maxlength=11')
        console.log(res.data)
            setMadLib({
                'blanks': res.data.blanks,
                'title': res.data.title,
                'value': res.data.value
            })
        }
        getMadLib();
    }, [])
    
    //handle form submit
    const onSubmit = () => {
        
    }
    
    return (
        <div className="MadLibForm">
            <form>
            <h1>MadLibForm</h1>
                {madLib ? madLib.blanks.map((word,idx) => {
                    return <input
                        key={ uuid() }
                        name={idx}
                        type="text"
                        placeholder={word}
                        onChange={handleChange}
                    />
                })
                    : 'Loading'}
            <button>Make some mad libs</button>
            </form>            
        </div>
    )
}

export default MadLibForm;