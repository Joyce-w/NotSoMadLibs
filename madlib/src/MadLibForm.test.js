import { render } from '@testing-library/react';
import MadLibForm from './MadLibForm';


//smoke test
test('renders learn react link', () => {
  render(<MadLibForm />);
});


//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<MadLibForm/>)
  expect(asFragment()).toMatchSnapshot();
})

//specialized testing
test("Test for form", () => {
    const { getByText } = render(<MadLibForm />)
    
    //get button text
    let button = getByText("Make some mad libs")
    expect(button).toBeInTheDocument();
})


