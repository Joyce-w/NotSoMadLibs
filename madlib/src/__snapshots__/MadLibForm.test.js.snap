// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MadLibForm"
  >
    <form
      class="MadLibForm-form"
    >
      <h1 />
      Loading
      <button>
        Make some mad libs
      </button>
    </form>
  </div>
</DocumentFragment>
`;
